src/web/page/pages.rs

```diff
@@ -0,0 +1,64 @@
+use super::TEMPLATE_DATA;
+use crate::docbuilder::Limits;
+use iron::{headers::ContentType, response::Response, status::Status, IronResult};
+use serde::Serialize;
+use tera::Context;
+
+/// The central trait that rendering pages revolves around, it handles selecting and rendering the template
+pub trait WebPage: Serialize + Sized {
```

Comment:
It's not actually on a trait definition, only in generic parameters; it's needed because `into_response` takes `self` by value so needs to know that it's `Sized`.